### PR TITLE
Fix #788

### LINKED ISSUE #788: chmod -+<octal> is broken

### PR DESCRIPTION
**Pull request by lucaotta**

### CODE REVIEW

File: src/chmod/chmod.rs [hunk 1]
```diff
@@ -40,7 +42,17 @@ pub fn uumain(args: Vec<String>) -> i32 {
     opts.optflag("R", "recursive", "change files and directories recursively");
     opts.optflag("h", "help", "display this help and exit");
     opts.optflag("V", "version", "output version information and exit");
-    // TODO: sanitize input for - at beginning (e.g. chmod -x testfile).  Solution is to add a to -x, making a-x
+    // sanitize input for - at beginning (e.g. chmod -x testfile). Remove
+    // the option and save it for later, after parsing is finished.
+    let re = Regex::new("^-{1}([rwx]|[0-7])").unwrap();
```
Comment (Arcterus) [inline src/chmod/chmod.rs]: I believe the regex should be `^-[rwxXstugo0-7]`.

Comment (lucaotta) [inline src/chmod/chmod.rs]: You're right, I've updated the pull request.

Comment (ebfe) [inline src/chmod/chmod.rs]: Do we really need to bring in the regex crate for this? #140

Comment (lucaotta) [inline src/chmod/chmod.rs]: Not really, it was already in place (maybe a leftover from a previous cleanup) and the resulting code is a bit easier to read. Without it, this is the best I can come up with my limited Rust-fu:

```rust
    for i in 0..args.len() {
        if let Some(first) = args[i].chars().nth(0) {
            if first != '-' {
                continue;
            }

            if let Some(second) = args[i].chars().nth(1) {
                match second {
                    'r' | 'w' | 'x' | 'X' | 's' | 't' | 'u' | 'g' | 'o' | '0' ... '7' => {
                        negative_option = Some(args.remove(i));
                        break;
                    },
                    _ => {}
                }
            }
        }
   }
```
What do you think?

Comment (lucaotta) [inline src/chmod/chmod.rs]: By the way, there are more dependencies that can be removed from chmod (eg. memchr, aho-corasick). I'll update the pull request when we agree on the regex part.

**Comment by ebfe** [comment]
ping @Arcterus

**Comment by lucaotta** [comment]
I've updated the code as per my previous comment and removed all unused dependencies from chmod. I've also changed the commit messages to honor the new guidelines.

File: src/chmod/chmod.rs [hunk 2]
```diff
@@ -40,7 +38,27 @@ pub fn uumain(args: Vec<String>) -> i32 {
     opts.optflag("R", "recursive", "change files and directories recursively");
     opts.optflag("h", "help", "display this help and exit");
     opts.optflag("V", "version", "output version information and exit");
-    // TODO: sanitize input for - at beginning (e.g. chmod -x testfile).  Solution is to add a to -x, making a-x
+    // sanitize input for - at beginning (e.g. chmod -x testfile). Remove
+    // the option and save it for later, after parsing is finished.
+    let mut negative_option = None;
+    for i in 0..args.len() {
+        if let Some(first) = args[i].chars().nth(0) {
```
Comment (Arcterus) [inline src/chmod/chmod.rs]: You can change `Some(first)` to `Some('-')` here so you can get rid of the check below.

**Comment by Heather** [comment]
@Arcterus @lucaotta I'm merging this because this pull request was inactive for month and correcting changes could be provided as separated PR

**Pull request merged**